common/lib/dependabot/git_commit_checker.rb

```diff
@@ -60,10 +60,10 @@ def pinned_ref_looks_like_version?
     end
 
     def pinned_ref_looks_like_commit_sha?
-      return false unless pinned?
-
       ref = dependency_source_details.fetch(:ref)
-      return false unless ref.match?(/^[0-9a-f]{6,40}$/)
+      return false unless ref&.match?(/^[0-9a-f]{6,40}$/)
+
+      return false unless pinned?
```

Comment:
This is performance: `pinned?` "does things" - but a regexp check is local and free!